src/Raven.Server/Documents/CollectionRunner.cs

```diff
@@ -86,7 +87,7 @@ public virtual Task<IOperationResult> ExecuteDelete(string collectionName, Colle
 
                             token.Delay();
 
-                            if (isAllDocs && IsHiLoDocument(document))
+                            if (isAllDocs && document.Id.StartsWith(HiLoHandler.RavenHiloIdPrefix, StringComparison.OrdinalIgnoreCase))
```

Comment:
Do we really need to be case insensitive here? This `StartsWith` overload can cause string allocations of `document.Id`

Reply:
Yes, we do need case insensitive in this. Document names are not case sensitive.